extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -536,10 +547,24 @@ export class AmpA4A extends AMP.BaseElement {
     //   - Chain cancelled => don't return; drop error
     //   - Uncaught error otherwise => don't return; percolate error up
     this.adPromise_ = viewerForDoc(this.getAmpDoc()).whenFirstVisible()
+        .then(() => {
+          checkStillCurrent(promiseId);
+          // See if experiment that delays request until slot is within
+          // renderOutsideViewport.
+          const resource = this.getResource();
+          // Within render outside viewport will not resolve if already within.
+          if (this.delayRequestEnabled && !resource.renderOutsideViewport()) {
+            return resource.whenWithinRenderOutsideViewport();
```

Comment:
This has a race condition with `#layoutCallback` that might come up if this is in the first viewport. If the layout process happens first, this will never resolve.